### PR TITLE
[api] fix log query from coreService

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

File: api/coreservice.go [hunk 1]
```diff
@@ -1267,57 +1221,78 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
 	return res
 }
 
-func (core *coreService) getLogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
+// GetLogsInBlock filter logs in the block x
+func (core *coreService) GetLogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
```
Comment (CoderZhi) [inline api/coreservice.go:1225]: LogsInBlock

File: api/coreservice.go [hunk 2]
```diff
@@ -1267,57 +1221,78 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
...
 
-// TODO: improve using goroutine
-func (core *coreService) getLogsInRange(filter *logfilter.LogFilter, start, end, paginationSize uint64) ([]*iotextypes.Log, error) {
-	if start > end {
-		return nil, errors.New("invalid start and end height")
-	}
-	if start == 0 {
-		start = 1
+// GetLogsInBlock filter logs among [start, end] blocks
+func (core *coreService) GetLogsInRange(filter *logfilter.LogFilter, start, end, maxPaginationSize uint64) ([]*iotextypes.Log, error) {
```
Comment (CoderZhi) [inline api/coreservice.go:1249]: LogsInRange

File: api/coreservice.go [hunk 3]
```diff
@@ -1267,57 +1221,78 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
...
-	logs := []*iotextypes.Log{}
 	// getLogs via range Blooom filter [start, end]
 	blockNumbers, err := core.bfIndexer.FilterBlocksInRange(filter, start, end)
 	if err != nil {
 		return nil, err
 	}
-	for _, i := range blockNumbers {
-		logsInBlock, err := core.getLogsInBlock(filter, i)
+	// TODO: improve getLogsInBlock using goroutine
+	logs := make([][]*iotextypes.Log, len(blockNumbers))
```
Comment (CoderZhi) [inline api/coreservice.go:1260]: why do you want to fetch all logs first?

File: api/grpcserver.go [hunk 4]
```diff
@@ -337,9 +338,29 @@ func (svr *GRPCServer) GetRawBlocks(ctx context.Context, in *iotexapi.GetRawBloc
...
+		return nil, status.Error(codes.InvalidArgument, "empty filter")
+	}
+	var (
+		ret []*iotextypes.Log
+		err error
+	)
+	switch {
+	case in.GetByBlock() != nil:
+		var blkHeight uint64
+		blkHeight, err = svr.coreService.dao.GetBlockHeight(hash.BytesToHash256(in.GetByBlock().BlockHash))
```
Comment (CoderZhi) [inline api/grpcserver.go:351]: we'd better not to use private variable of coreService

Comment (Liuhaai) [reply api/grpcserver.go:352]: We could add a new func `GetBlockHeight()` in another pr

File: api/coreservice.go [hunk 5]
```diff
@@ -1267,59 +1221,81 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
...
 
-// TODO: improve using goroutine
-func (core *coreService) getLogsInRange(filter *logfilter.LogFilter, start, end, paginationSize uint64) ([]*iotextypes.Log, error) {
-	if start > end {
-		return nil, errors.New("invalid start and end height")
-	}
-	if start == 0 {
-		start = 1
+// LogsInRange filter logs among [start, end] blocks
+func (core *coreService) LogsInRange(filter *logfilter.LogFilter, start, end, maxPaginationSize uint64) ([]*iotextypes.Log, error) {
```
Comment (dustinxie) [inline api/coreservice.go:1249]: `paginationSize` is correct/clear, no need for `max`

**Review by dustinxie — APPROVED** [review]

File: api/grpcserver.go (hunk 4 above)
Comment (CoderZhi) [reply api/grpcserver.go:352]: it should be LogsByBlockHash

**Review by CoderZhi — APPROVED** [review]

File: api/grpcserver.go (hunk 4 above)
Comment (Liuhaai) [reply api/grpcserver.go:352]: BlockMetaByHash() also shares the dao.GetBlockHeight() func

**Pull request merged**